File: core/src/main/java/org/dreamexposure/discal/core/enums/announcement/AnnouncementModifier.java
package org.dreamexposure.discal.core.enums.announcement;

public enum AnnouncementModifier {
    BEFORE, DURING, END;

    public static boolean isValid(final String _value) {
        return "BEFORE".equalsIgnoreCase(_value) || "B4".equalsIgnoreCase(_value) || "DURING".equalsIgnoreCase(_value)
            || "END".equalsIgnoreCase(_value);
    }

    public static AnnouncementModifier fromValue(final String value) {
        switch (value.toUpperCase()) {
            case "DURING":
                return DURING;
            case "END":
                return END;
            default:
                return BEFORE;
        }
    }
}

File: core/src/main/kotlin/org/dreamexposure/discal/core/enums/announcement/AnnouncementModifier.kt
package org.dreamexposure.discal.core.enums.announcement

enum class AnnouncementModifier {
    BEFORE, DURING, END;

    companion object {
        fun isValid(value: String): Boolean {
            return value.equals("BEFORE", true)
                    || value.equals("B4", true)
                    || value.equals("DURING", true)
                    || value.equals("END", true)
        }

        fun fromValue(value: String): AnnouncementModifier {
            return when (value.uppercase()) {
                "DURING" -> DURING
                "END" -> END
                else -> BEFORE
            }
        }
    }
}
